modules/images/webp-uploads/load.php

```diff
@@ -684,6 +684,135 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
 
+/**
+ * Adds sources to metadata for an attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $metadata              Metadata of the attachment.
+ * @param array  $valid_mime_transforms List of valid mime transforms for current image mime type.
+ * @param string $file                  Path to original file.
+ * @param array  $main_images           Path of all main image files of all mime types.
+ * @param array  $subsized_images       Path of all subsized image file of all mime types.
+ *
+ * @return array                        Metadata with sources added.
+ */
+function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $file, $main_images, $subsized_images ) {
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		// Add sources to original image metadata.
+		$image_file = $main_images[ $targeted_mime ]['path'];
+
+		if ( ! file_exists( $image_file ) ) {
+			continue;
+		}
+
+		$metadata['sources'][ $targeted_mime ] = array(
+			'file'     => wp_basename( $image_file ),
+			'filesize' => filesize( $image_file ),
+		);
+
+		foreach ( $metadata['sizes'] as $size => $size_details ) {
+			// Add sources to resized image metadata.
+			$image_file = $original_directory . '/' . $subsized_images[ $targeted_mime ][ $size ]['file'];
+
+			if ( ! file_exists( $image_file ) ) {
+				continue;
+			}
+
+			$metadata['sizes'][ $size ]['sources'][ $targeted_mime ] = array(
+				'file'     => wp_basename( $image_file ),
+				'filesize' => filesize( $image_file ),
+			);
+		}
+	}
+
+	return $metadata;
+}
+
+/**
+ * Creates additional image formats when original image is edited.
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( null !== $override ) {
+		return $override;
+	}
+
+	if ( empty( webp_uploads_get_upload_image_mime_transforms()[ $mime_type ] ) ) {
```

Comment:
Nit-pick, but WP core standards suggest to avoid array access directly on a function return value, so let's assign this to a variable first.

I'd also suggest we pass this to the closure below via `use`, so that we don't have to look up the exact same thing again in that closure.